src/main/java/com/mparticle/kits/IterableKit.java

```diff
@@ -0,0 +1,86 @@
+package com.mparticle.kits;
+
+import android.app.Activity;
+import android.content.Context;
+import android.os.Bundle;
+
+import com.mparticle.DeepLinkResult;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ *
+ * This is an mParticle kit, used to extend the functionality of mParticle SDK. Most Kits are wrappers/adapters
+ * to a 3rd party SDK, primarily used to map analogous public mParticle APIs onto a 3rd-party API/platform.
+ */
+public class IterableKit extends KitIntegration implements KitIntegration.ActivityListener {
+
+   private String deeplinkUrl;
+
+    @Override
+    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, Context context) {
+        return null;
+    }
+
+    @Override
+    public String getName() {
+        return "Iterable";
+    }
+
+    @Override
+    public List<ReportingMessage> setOptOut(boolean optedOut) {
+        ReportingMessage optOutMessage = new ReportingMessage(this, ReportingMessage.MessageType.OPT_OUT, System.currentTimeMillis(), null);
```

Comment:
If you're not going to forward the opt-out event, you can just return null here rather than creating an object.